app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelTop.tsx

```diff
@@ -106,6 +107,20 @@ const ConnectionPanelTop = ({
 	const serverIsNotDuplicate = server.connectedState !== ConnectedState.DUPLICATE;
 	const isConfigPage = moduleKey && moduleKey === CONFIG_PAGE;
 
+	const buttonAndIconAppearance = disconnectError ? 'danger' : 'warning';
+	const modalTitleMessage =
+		disconnectError ? 'An error occurred while deleting your server' : 'Disconnect this Jenkins server?';
+	const modalBodyMessage = disconnectError ? [
```

Comment:
I feel like something like this would work a bit better than the arrays:
```
const modalBodyMessage = disconnectError ? 
    (
        <>Something went wrong while disconnecting 
        <strong key={server.name}>{serverToDisconnect?.name}</strong>, 
        please try again.</>
    ) 
    : 
    (
        <>Are you sure that you want to disconnect your Jenkins server, 
        <strong key={server.name}>{serverToDisconnect?.name}</strong>? 
        This means that you disconnect all associated Jenkins jobs, and will have to add a new server in Jira if you ever want to reconnect.</>
    );
    ```

Reply:
Also, a couple of questions here:
1. Are you doing anything with the key?
2. If the server name is optional, what happens when it doesn't appear? I don't think it should be optional.

Reply:
That won't work for the typing defined in JenkinsModal `body: (string | React.ReactElement<any>)[]`. We could always update it but probably to tackle that after we rip out the existing code to prevent less changes on that end.

Reply:
> Are you doing anything with the key?

Good catch. That was something I meant to remove and forgot. server.name isn't optional though. This component only gets rendered if there are servers and the name is a required property.

Reply:
No problem r.e. the array. If it's required, we should be able to remove the `serverToDisconnect?` conditional